charts/consul/templates/connect-inject-clusterrole.yaml

```diff
@@ -144,4 +147,27 @@ rules:
   - get
   - patch
   - update
+- apiGroups:
+    - apps
+  resources:
+    - deployments
+  verbs:
+    - create
+    - get
+    - list
+    - update
+    - watch
+- apiGroups:
+    - core
+  resources:
+    - services
+  verbs:
+    - watch
+    - list
+- apiGroups: [ "" ]
+  resources: [ "secrets" ]
```

Comment:
Should we put this behind a "if gateways.enabled"?

Reply:
No, we decided to just have API Gateways gated behind "connectInject.enabled".